src/Computation/Values/And.cs

```diff
@@ -34,17 +34,18 @@ private static IValue Create(IValue left, ConstantUnsigned right)
         return left switch
         {
             IConstantValue l => l.AsUnsigned().And(right),
-            And l => new And(l._left, Create(l._right, right)),
+            And l => Create(l._left, Create(l._right, right)),
             _ => new And(left, right)
         };
     }
 
     public static IValue Create(IValue left, IValue right)
     {
-        return right is IConstantValue r
-            ? ShortCircuit(left, r.AsUnsigned())
-            : left is IConstantValue l
-                ? ShortCircuit(right, l.AsUnsigned())
-                : new And(left, right);
+        return (left, right) switch
```

Comment:
I also find the style more readable and amenable to any additional cases we want to treat, so I've converted them all. Can revert if you don't agree.